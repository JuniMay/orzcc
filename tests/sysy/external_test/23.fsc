// polybench-c-4.2 linear-algebra/kernels/mvt
#include <stdio.h>
const int N = 2000;

float A[2000][2000] = {};
float x1[2000] = {};
float x2[2000] = {};
float y_1[2000] = {};
float y_2[2000] = {};

void init() {
    int i = 0, j = 0;

    while (i < N) {
        x1[i] = (i % N) / N;
        x2[i] = ((i + 1) % N) / N;
        y_1[i] = ((i + 3) % N) / N;
        y_2[i] = ((i + 4) % N) / N;

        j = 0;
        while (j < N) {
            A[i][j] = (i * j % N) / N;
            j = j + 1;
        }
        i = i + 1;
    }
}

void kernel_mvt() {
    int i = 0, j = 0;

    i = 0;
    while (i < N) {
        j = 0;
        while (j < N) {
            x1[i] = x1[i] + A[i][j] * y_1[j];
            j = j + 1;
        }
        i = i + 1;
    }

    i = 0;
    while (i < N) {
        j = 0;
        while (j < N) {
            x2[i] = x2[i] + A[j][i] * y_2[j];
            j = j + 1;
        }
        i = i + 1;
    }
}

int main() {
    init();
    kernel_mvt();
    printf("%a",x1[0] + x2[0] + x1[1] + x2[1]);
    return 0;
}

