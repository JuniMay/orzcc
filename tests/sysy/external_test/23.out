0x1.4c35d4p+10
0
